app/service/parsing_svc.py

```diff
@@ -21,14 +55,11 @@ def __init__(self):
         for result in [r for r in results if not r['parsed']]:
             parser = await data_svc.get('core_parser', dict(ability=result['link']['ability']))
             if parser and result['link']['status'] == 0:
-                if parser[0]['name'] == 'json':
-                    matched_facts = parsers.json(parser[0], b64decode(result['output']).decode('utf-8'), self.log)
-                elif parser[0]['name'] == 'line':
-                    matched_facts = parsers.line(parser[0], b64decode(result['output']).decode('utf-8'), self.log)
-                elif parser[0]['name'] == 'mimikatz':
-                    matched_facts = mimikatz_parser.mimikatz(b64decode(result['output']).decode('utf-8'), self.log)
-                else:
-                    matched_facts = parsers.regex(parser[0], b64decode(result['output']).decode('utf-8'), self.log)
+                if parser[0]['name'] in self.parsers:
+                    matched_facts = self.parsers[parser[0]['name']].parse(parser[0], b64decode(result['output']).decode('utf-8'))
+                else:                
+                    self.log.error('Unable to find parser %s' % parser[0]['name'])
+                    matched_facts = []
```

Comment:
instantiate matched facts at the top and remove the else block.
if the parser isn't there a keyerror will be thrown on line 59.

also might be good to handle the possible key error exception

Reply:
Wouldn't that make the code harder to debug?  If I catch a KeyError instead of the if/else, potentially there can be many locations where the KeyError was generated from.  Unless you are suggesting wrapping the self.parsers call around try/catch?